apps/studio/components/interfaces/Organization/BillingSettings/BillingAddress/BillingAddressDialog.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from 'react'
+import { Button, Dialog, DialogContent } from 'ui'
+import { Label_Shadcn_ as Label } from 'ui'
+import BillingAddressForm from './BillingAddressForm'
+import { Pencil } from 'lucide-react'
+
+import { useCheckPermissions } from 'hooks/misc/useCheckPermissions'
+import { useOrganizationCustomerProfileQuery } from 'data/organizations/organization-customer-profile-query'
+import { PermissionAction } from '@supabase/shared-types/out/constants'
+import NoPermission from 'components/ui/NoPermission'
+import ShimmeringLoader from 'components/ui/ShimmeringLoader'
+import AlertError from 'components/ui/AlertError'
+
+interface BillingAddressDialogProps {
+  slug: string
+}
+
+const BillingAddressDialog = ({ slug }: BillingAddressDialogProps) => {
```

Comment:
might be better to rename this modal to something like BillingCustomerDataDialog or so, as it will cover address + tax id